my-app/src/components/Header.tsx

```diff
@@ -1,39 +1,36 @@
-import React, {useEffect, useState} from 'react';
+import React, {useContext, useEffect, useState} from 'react';
 import styled from "styled-components";
-import {Link, useNavigate} from "react-router-dom";
+import {Link} from "react-router-dom";
 import Container from "@mui/material/Container";
 import {Button} from "@mui/material";
-import {useLocalStorage} from "../helpers/functions";
+import {LogoutContext} from "../pages/authentication/AuthContext";
 
 
 export const Header = () => {
-    const [token, setToken] = useState<any>('')
-    useEffect(()=>{
-        setToken(localStorage.getItem('token'))
-    },[])
+    const logic = useContext(LogoutContext)

```

Comment:
logic is a bit confusing name, just get functions that you need from this hook. 

like `const { setToken } = useContext(LogoutContext)`

Reply:
I find it cool, useLogicState allows getting and passing to useState,  https://dev.to/svehla/react-typed-state-management-under-10-lines-of-code-1347

Reply:
![Screenshot 2022-11-13 at 19 40 17](https://user-images.githubusercontent.com/38957865/201538558-7397cab2-b93d-46a3-8b68-59bf62d85984.png)

I'm just saying that you can get functions out of this hook directly and you won't need use dot notation to use it. 

in your code you will just use `setToken()` instead of `logic.setToken()`

Reply:
fixed

Reply:
fixed